src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -676,16 +582,29 @@ protected WorkerPool(ThreadFactory threadFactory, int poolSize) {
       }
       
       poolSizeChangeLock = new Object();
-      workersLock = new Object();
-      availableWorkers = new ArrayDeque<Worker>(poolSize);
-      waitingForWorker = false;
-      currentPoolSize = 0;
+      idleWorker = new AtomicReference<Worker>(null);
+      currentPoolSize = new AtomicInteger(0);
       
       this.threadFactory = threadFactory;
       this.maxPoolSize = poolSize;
+      this.workerTimedPark = false;
       shutdownStarted = new AtomicBoolean(false);
       shutdownFinishing = false;
     }
+    
+    /**
+     * Starts the pool, constructing the first thread to start consuming tasks (and starting other 
+     * threads as appropriate).  This should only be called once, and can NOT be called concurrently.
+     * 
+     * @param queueManager QueueManager to source tasks for execution from
+     */
+    public void start(QueueManager queueManager) {
```

Comment:
This is kind of crappy because it creates a circular dependency between this and QueueManager.  That's why it is provided after the start.  QueueManager needs a handle on the pool to notify of queue updates.  Pool needs a handle of the queueManager to poll tasks for execution.

The only solution I see is to combine the classes, or to put one of them into the parent class, but that seems kind of ugly too since they really have a nice separation of needs right now.